rezasm-source/rezasm-core/src/parser/line.rs

```diff
@@ -53,16 +56,34 @@ impl Line {
         }
 
         let mut arguments: Vec<ArgumentType> = Vec::new();
-        for argument in args_out {
-            arguments.push(match argument.get_target(word_size) {
-                Ok(arg) => arg,
-                Err(error) => return Err(error),
-            });
-        }
+        let instruction_retrieved =
+            match instruction_registry::get_instruction(instruction, args_out.len()) {
+                Ok(instruction) => instruction,
+                Err(e) => return Err(e),
+            };
 
-        let matching_instruction = get_instruction(instruction, &arguments)?;
+        if instruction_retrieved.get_types().len() != args_out.len() {
+            return Err(ParserError::InvalidInstructionError(instruction.clone()));
+        }
 
-        Ok(Line::Instruction(matching_instruction, arguments))
+        for (argument, type_of) in args_out.iter().zip(instruction_retrieved.get_types()) {
+            if type_of == &TypeId::of::<&mut InputTarget>() {
+                arguments.push(argument.get_input_target(word_size)?);
+            } else if type_of == &TypeId::of::<&mut InputOutputTarget>() {
+                arguments.push(match argument.get_input_output_target(word_size) {
+                    Ok(s) => s,
+                    Err(e) => match e {
+                        ParserError::InternalError => {
+                            return Err(ParserError::InvalidArgumentsError(instruction.clone()))
+                        }
+                        _ => return Err(e),
+                    },
+                });
+            } else {
+                return Err(ParserError::InvalidArgumentsError(instruction.to_string()));
```

Comment:
We may want to change this error or create a new error to be more descriptive by including which token was of the incorrect type.